File: python/0107.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def __init__(self):
        self.depth=0
    def levelOrderBottom(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        a=[]
        def go(r,dep):
            if not r:return
            self.depth=max(self.depth,dep)
            a.append((r.val,dep,len(a)))
            go(r.left,dep+1)
            go(r.right,dep+1)
        go(root,0)
        depth=self.depth
        ans=[[]for i in range(depth+1)]
        for i in a:
            ans[i[1]].append(i)
        for i in range(depth+1):
            ans[i]=sorted(ans[i],key=lambda x:x[2])
            ans[i]=map(lambda x:x[0],ans[i])
        ret=[]
        for i in ans:
            if i:
                ret.append(i)
        return ret[::-1]

File: rust/src/bin/0107.rs
// Definition for a binary tree node.
#[derive(Debug, PartialEq, Eq)]
pub struct TreeNode {
    pub val: i32,
    pub left: Option<Rc<RefCell<TreeNode>>>,
    pub right: Option<Rc<RefCell<TreeNode>>>,
}

impl TreeNode {
    #[inline]
    pub fn new(val: i32) -> Self {
        TreeNode {
            val,
            left: None,
            right: None,
        }
    }
}
struct Solution;
use std::rc::Rc;
use std::cell::RefCell;

impl Solution {
    pub fn level_order_bottom(root: Option<Rc<RefCell<TreeNode>>>) -> Vec<Vec<i32>> {
        let mut sta = vec!();
        sta.push(&root);
        let mut ans = vec!();
        while !sta.is_empty() {
            let mut next_level = vec!();
            let mut cur_level = vec!();
            while !sta.is_empty() {
                let now = sta.remove(0);
                if now.is_none() {
                    continue;
                }
                unsafe {
                    let node = &*now.as_deref().unwrap().as_ptr();
                    cur_level.push(node.val);
                    next_level.push(&node.left);
                    next_level.push(&node.right);
                }
            }
            if cur_level.len() > 0 {
                ans.push(cur_level);
            }
            sta = next_level;
        }
        ans.reverse();
        ans
    }
}

fn main() {
    println!("good");
}